### DISCUSSION TITLE
Handling cases where most of the site/domain already has a native dark mode except for specific subdomains?

### DISCUSSION THREAD

**Discussion by X-m7** [darkreader/darkreader#6965] [Ideas]
I have the site list set to "not invert listed", and in there I have github.com, since it's got a native dark mode already. However, securitylab.github.com doesn't have that, so when I try to use the Alt+Shift+A shortcut or the toggle current site button to try to make it dark all that happens is that it gets added to the list, which of course does nothing, the site stays light.

So, would it be practical to make the toggle current site button at least temporarily do just that if adding the site to the list will not have an effect? The extension seems to know (or be able to know) that already, the toggle current site button just keeps the cross (so it's in the off state and stays like that) when I click it for securitylab.github.com, while if I switch to the site list section I can see it getting added and removed as I click it.

There are ways around the issue, like enabling the extension for github.com, which will then also enable it on securitylab.github.com, or switching the site list to "invert listed", but I think it would be great if the toggle just works as a toggle no matter the situation with the list. I'd understand if this is considered too much of an edge case to be considered though.

**Comment by Gusted** [comment]
> So, would it be practical to make the toggle current site button at least temporarily do just that if adding the site to the list will not have an effect? The extension seems to know (or be able to know) that already, the toggle current site button just keeps the cross (so it's in the off state and stays like that) when I click it for securitylab.github.com, while if I switch to the site list section I can see it getting added and removed as I click it.

So that's a bit of a problem with our current matching system. As that is not possible. It will be nice if you can follow issues like #373 where this is discussed and a more nicer matching is introduced(which is able to handle this case).



> There are ways around the issue, like enabling the extension for github.com, which will then also enable it on securitylab.github.com, or switching the site list to "invert listed", but I think it would be great if the toggle just works as a toggle no matter the situation with the list. I'd understand if this is considered too much of an edge case to be considered though.

You could add `^github.com` which will mean that darkreader is enabled on all subdomains.

**Reply by X-m7** [reply]
Oh, I had no idea what sort of keywords I'd need to use to search through the issues list, hence this post. The `^github.com` trick works perfectly, thanks for the tip!
